src/reducers/starWarsReducer.js

```diff
@@ -1,13 +1,40 @@
-import /* we need our action types here*/ "../actions";
+import /* we need our action types here*/  {FETCHING ,  SUCCESS  ,   FAILURE }       from "../actions";
 const initialState = {
-  characters: []
+  characters: [],
+  fetching:false,
+  error:'',
   // Array characters, Boolean fetching, null error.
 };
 export const charsReducer = (state = initialState, action) => {
   switch (action.type) {
     // Fill me in with the important reducers
     // action types should be FETCHING, SUCCESS and FAILURE
     // your switch statement should handle all of these cases.
+
+    case FETCHING:
+    return {
+      ...state,
+      fetching:true,
+      error:''
+    }
+    
+    case SUCCESS:
+    return {
+      ...state,
+      fetching:false,
+      characters:[...state.characters,...action.payload],
+      error:''
```

Comment:
When there is no error, keep it set to `null` instead of an empty string `""` - `null` states more declaratively that there is no error, as opposed to having an 'error' that is composes of empty text.